model/project_parser.go

```diff
@@ -1019,6 +1019,26 @@ func createIntermediateProject(yml []byte, unmarshalStrict bool) (*ParserProject
 	if p.Functions == nil {
 		p.Functions = map[string]*YAMLCommandSet{}
 	}
+
+	// Special case: skip priority capping for the mongo-release project.
+	if utility.FromStringPtr(p.Identifier) == "mongo-release" {
+		return &p, nil
+	}
+
+	// Cap priority values at MaxConfigSetPriority
+	for i := range p.Tasks {
+		if p.Tasks[i].Priority > MaxConfigSetPriority {
+			p.Tasks[i].Priority = MaxConfigSetPriority
+		}
+	}
+	for i := range p.BuildVariants {
+		for j := range p.BuildVariants[i].Tasks {
+			if p.BuildVariants[i].Tasks[j].Priority > MaxConfigSetPriority {
+				p.BuildVariants[i].Tasks[j].Priority = MaxConfigSetPriority
+			}
+		}
+	}
```

Comment:
Is it possible to add some tests for this change? I'm not sure why the particular change from the solo `if` statement to these larger blocks, it seems like we should be able to keep the logic consolidated to some degree

Reply:
the reason I moved the priority capping to here was because it wasn't possible to cap bvts in the same place in the previous location